src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -4321,7 +4527,22 @@ the (Stream step state) = go state
 
 {-# INLINE runFold #-}
 runFold :: (Monad m) => Fold m a b -> Stream m a -> m b
-runFold (Fold step begin done) = foldlMx' step begin done
+runFold (Fold fstep begin done) (Stream step state) =
```

Comment:
We can call this `foldOnce` vs `foldMany`, this just the same as `foldMany` except that we fold only once. In the context of terminating folds its better to make the distinction.